spark/v3.1/spark/src/main/scala/com/netease/arctic/spark/sql/execution/ExtendedArcticStrategy.scala

```diff
@@ -103,6 +106,9 @@ case class ExtendedArcticStrategy(spark: SparkSession) extends Strategy with Pre
       CreateArcticTableAsSelectExec(catalog, ident, parts, query, planLater(query), planLater(validateQuery),
         props, writeOptions, ifNotExists) :: Nil
 
+    case d@AlterArcticTableDropPartition(r: ResolvedTable, _, _, _, _) =>
+      AlterArcticTableDropPartitionExec(r.table, d.parts, d.retainData) :: Nil
+
```

Comment:
I think here we use 
```
case d @ AlterTableDropPartition if isArcticTable(t) => 
  AlterArcticTableDropPartitionExec
``` 
should be better. 


The logical command `AlterArcticTableDropPartition` seems useless.